src/Reader.ts

```diff
@@ -0,0 +1,442 @@
+import Publication from "./model/Publication";
+import { UserSettings } from "./model/user-settings/UserSettings";
+import AnnotationModule from "./modules/AnnotationModule";
+import BookmarkModule from "./modules/BookmarkModule";
+import TextHighlighter from "./modules/highlight/TextHighlighter";
+import TimelineModule from "./modules/positions/TimelineModule";
+import ContentProtectionModule from "./modules/protection/ContentProtectionModule";
+import SearchModule from "./modules/search/SearchModule";
+import TTSModule from "./modules/TTS/TTSModule";
+import { TTSSettings } from "./modules/TTS/TTSSettings";
+import IFrameNavigator, {
+  ReaderConfig,
+  UpLinkConfig,
+} from "./navigator/IFrameNavigator";
+import LocalAnnotator from "./store/LocalAnnotator";
+import LocalStorageStore from "./store/LocalStorageStore";
+
+/**
+ * Gets the HTML element and throws an error if it doesn't exist
+ */
+function getElement(id: string): HTMLElement {
+  const elem = document.getElementById(id);
+  if (!elem) throw new Error(`Missing required element with ID: ${id}`);
+  return elem;
+}
+
+/**
+ * A class that, once instantiated using the public `.build` method,
+ * is the primary interface into the D2 Reader.
+ *
+ * @TODO :
+ *  - DEV logger
+ */
+export default class Reader {
+  private constructor(
+    readonly settings: UserSettings,
+    readonly ttsSettings: TTSSettings,
+    readonly navigator: IFrameNavigator,
+    readonly highlighter: TextHighlighter,
+    readonly bookmarkModule: BookmarkModule,
+    readonly annotationModule?: AnnotationModule,
+    readonly ttsModule?: TTSModule,
+    readonly searchModule?: SearchModule,
+    readonly contentProtectionModule?: ContentProtectionModule,
+    readonly timelineModule?: TimelineModule
+  ) {}
+  /**
+   * The async builder.
+   * @TODO :
+   *  - browser support section
+   *  - Code that updates/changes the config before we start using it
```

Comment:
These are two big things from the old `load` function in `index.ts` that I have yet to build into this.